src/controllers/product.ts

```diff
@@ -0,0 +1,47 @@
+import {Request, Response} from "express";
+import {getUserById} from "../services/user";
+import {returnUnauthorized} from "../utils/return-unauthorized";
+import {ID_HEADER_NAME} from "./user";
+import {getProduct, getProducts} from "../services/product";
+
+export const productsHandler = (req: Request, res: Response) => {
```

Comment:
According to your code you're getting one product by ID, so you'd better name you function accordingly, and "handler" part is optional (naming "handler" is usually used for AWS lambdas).
```suggestion
export const getProductById = (req: Request, res: Response) => {
```